tests/generators/test_cpp.py

```diff
@@ -278,6 +279,32 @@ def test_when_names_invalid_then_warnings_are_logged(
         assert a_capturing_logger.handlers[0].messages == warnings_exotic_values
 
 
+def test_cpp_project(generated_path, tmp_path_factory, cpp_project_dir):
```

Comment:
You know on further thought, I guess it is pretty convenient to be able to test all of traitgen from one place, will close this.

Reply:
Re-opening this, since I notice our CI build matrix is based on Python version, so currently we're re-running all the C++ tests pointlessly.

We can 
* Leave it like it is - though the C++ tests do take a long time to run.
* Leave C++ tests in `pytest`, but conditionally run the C++ tests once per platform.
* Run the C++ tests independently. This adds some duplicated boilerplate.

We also need to consider the C++ linters, which currently run all the time, pointlessly (and likely to fail on non-Ubuntu) and severely increase the test time.

I'm leaning toward pulling the C++ tests out of `pytest`.

Reply:
Explicitly skipped "ctest" marked tests in CI in 808ac0b1